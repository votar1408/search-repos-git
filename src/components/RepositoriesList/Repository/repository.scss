@import '../../../assets/scss/variable.scss';

.repository {
    display: flex;
    padding: 10px;
    border-bottom: 1px solid $color-grey;
}

.repository__block {
    display: flex;
    align-items: center;
}

.repository__block_title {
    font-weight: bold;
    margin-right: 6px;
    font-size: 18px;
}

.repository__block_description {
    min-width: 200px;
    max-width: 200px;
    margin-right: 6px;
    white-space: nowrap;
    overflow: hidden;
    font-size: 16px;
    text-overflow: ellipsis;
}

.repository__block_logo {
    width: 34px;
    height: 34px;
    border-radius: 34px;
    margin-right: 6px;
    background-color: white;
}
