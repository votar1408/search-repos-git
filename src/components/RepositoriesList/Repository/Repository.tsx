import './repository.scss';

import React, {FC} from 'react';

import {texts} from '../../../consts';
import {RepositoryServerInfo} from '../../../interfaces/ServerData';

interface RepositoryProps {
    item: RepositoryServerInfo;
}

export const Repository: FC<RepositoryProps> = ({
    item: {
        id,
        name,
        description,
        owner: {login, avatar_url}
    }
}: RepositoryProps) => {
    return (
        <li className="repository">
            <div className="repository__block">
                <span className="repository__block_title">{texts.name}</span>
                <span className="repository__block_description">{name}</span>
            </div>
            <div className="repository__block">
                <span className="repository__block_title">{texts.description}</span>
                <span className="repository__block_description">{description}</span>
            </div>
            <div className="repository__block">
                <span className="repository__block_title">{texts.owner}</span>
                {avatar_url ? <img className="repository__block_logo" src={avatar_url} alt="logo" /> : null}
                <span className="repository__block_description">{login}</span>
            </div>
        </li>
    );
};
