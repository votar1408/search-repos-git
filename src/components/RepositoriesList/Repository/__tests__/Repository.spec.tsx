import React from 'react';

import {
    shallow,
    ShallowWrapper
} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Repository} from '../Repository';

describe('Repository', () => {
    const repositoryData = {
        id: 1,
        name: 'some name',
        description: 'some description',
        owner: {login: 'some id', avatar_url: 'some url'}
    };
    const wrapper: ShallowWrapper = shallow(<Repository item={repositoryData} />);

    it('should render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should contains 3 <span/>', () => {
        expect(wrapper.find('div[data-id="repository-block"]').length).toEqual(3);
    });
});
