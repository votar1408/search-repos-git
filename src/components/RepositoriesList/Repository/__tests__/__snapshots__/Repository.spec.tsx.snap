// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository should render correctly 1`] = `
<li
  className="repository"
>
  <div
    className="repository__block"
    data-id="repository-block"
  >
    <span
      className="repository__block_title"
    >
      Name:
    </span>
    <span
      className="repository__block_description"
    >
      some name
    </span>
  </div>
  <div
    className="repository__block"
    data-id="repository-block"
  >
    <span
      className="repository__block_title"
    >
      Description:
    </span>
    <span
      className="repository__block_description"
    >
      some description
    </span>
  </div>
  <div
    className="repository__block"
    data-id="repository-block"
  >
    <span
      className="repository__block_title"
    >
      Owner:
    </span>
    <img
      alt="logo"
      className="repository__block_logo"
      src="some url"
    />
    <span
      className="repository__block_description"
    >
      some id
    </span>
  </div>
</li>
`;
